flyteplugins/go/tasks/pluginmachinery/k8s/plugin.go

```diff
@@ -45,6 +55,8 @@ type PluginProperties struct {
 	// override that behavior unless the resource that gets created for this plugin does not consume resources (cluster's
 	// cpu/memory... etc. or external resources) once the plugin's Plugin.GetTaskPhase() returns a terminal phase.
 	DisableDeleteResourceOnFinalize bool
+	// Specifies how errors are aggregated
+	ErrorAggregationStrategy ErrorAggregationStrategy
```

Comment:
Why is this a property of the plugin? Isn't the EarliestError Strategy already backward compatible? 
I think if there is anything a plugin needs to optionally configure, it's the pattern of the error file names

Reply:
It is backward compatible, yes. But the RFC called for error strategies, so different ones can be implemented for different plugins. Are you thinking of making it the default? 

@fg91  What do you think about this?

Reply:
It's not clear to me what those other implementations may look like... I usually like to wait for a second/third option before adding an enum/config... If we don't know what implementations we may want to add here, I would say yes let's make it the default and allow plugins to configure the file name pattern...

Reply:
Let's break this into 2 revision requests:

1. Do not have an error aggregation strategy enum and make the new implementation, which is backward compatible, the default: I have no objections to this, but then I have the least amount of knowledge about the Flyte ecosystem here. The RFC called out for error aggregation strategies, in anticipation of more plugins making use of this new customization option. Also, not changing the default and using the multi-error file strategy where it is needed (PyTorch plugin for now) seems a bit safer then starting to adopt it for all. I need more feedback here from the involved parties. Pinging @fg91 and @eapolinario as they participated in the RFC.

2. Configuring file name patterns: What benefits do you see here? It will further complicate flytekit side error file creation. If there is a motivation, we can do it.

Reply:
> It's not clear to me what those other implementations may look like... I usually like to wait for a second/third option before adding an enum/config...

In the [RFC](https://github.com/flyteorg/flyte/pull/5598/files) (not accepted yet so all of this is up for discussion), we propose that the currently existing logic with a single `error.pb` file is denoted a strategy as well with enum index `0` (default behaviour). "Earliest" would then be index `1`.

---

> Why is this a property of the plugin? Isn't the EarliestError Strategy already backward compatible?

The RFC currently states this about backwards compatibility:

> We propose that the new `MultiErrorFileRemoteFileOutputReader` falls back to reading the `error.pb` (behaviour of the default `RemoteFileOutputReader`) if no `error-<pod-name>.pb` files are found in order to solve the problem of backwards compatibility:

> * If flytekit uses a version that supports multiple error files but the backend does not yet, `pyflyte-execute` will not upload multiple error files for distributed tasks since the `FLYTE_INTERNAL_DIST_ERROR_STRATEGY` environment variable will not be set.
> * If flytekit uses an older version that does not support multiple error files while the backend does, a single error file will be uploaded despite `FLYTE_INTERNAL_DIST_ERROR_STRATEGY` being set. The output reader will, however, fall back to reading the single `error.pb`.

I think plugins need to configure in their properties whether they make use of the default single `error.pb` strategy (`0` in the enum) or the "earliest of multiple" strategy (`1` in the enum) despite this backwards compatibility because most tasks will need the default strategy. Always first checking if multiple errors files exist to then fall back to the single one will make a lot of unnecessary requests to blob storage since for certain plugins (in particular normal python task!) we already know that only a single file will be written.

I'd only use this fallback in plugins that use multiple error files to account for version mismatches between flytekit and backend.

---

> I think if there is anything a plugin needs to optionally configure, it's the pattern of the error file names

In an earlier version of the RFC we included configurable file name patterns but for simplicity decided to propose to add an `/errors` "directory" (being fully aware that buckets are not a filesystem) to the raw output prefix bucket uri. Any files in this error bucket would be considered potential error files.

---

Please let me know if you have any questions or disagree about this @EngHabu. As said above, the RFC is still open and I'm happy to discuss and change it.

Reply:
> Always first checking if multiple errors files exist to then fall back to the single one will make a lot of unnecessary requests to blob storage since for certain plugins (in particular normal python task!) we already know that only a single file will be written.

The overhead will be a single additional list call, which will return the single error.pb file.

Reply:
Yes, a single additional call. But pytorch (or other distributed tasks) generally are not the majority of tasks and for a normal python task this is never required. I'd argue we should avoid this unnecessary call if we already know it's void.